Framework/Beamline/src/DetectorInfo.cpp

```diff
@@ -305,6 +305,18 @@ Eigen::Vector3d DetectorInfo::samplePosition() const {
   return m_componentInfo->samplePosition();
 }
 
+bool DetectorInfo::isRectangularDetectorPixel(const size_t index) const {
```

Comment:
I am wondering if we should make this a more general concept? `RectangularDetector` is a special case, but I suppose a lot of client code might be interested more generally in whether this is a regular/structured assembly?

Also, I think the naming is misleading, it seems to imply that the pixel is rectangular, the actual meaning (it is part of a rectangular/regular assembly of pixels) is only clear to people familiar with the old instrument code.

Reply:
I'd be tempted to remove it. What do you think? We don't actually need it for this current work. I do however like `ComponentInfo::isRectangularBank`, that's required for this work, and generally something I think that would be useful.

Reply:
I think `isRectangularBank` is also a bit narrow. I do not strongly object having it now, but maybe we can think of this concept could/should be made more general to support, e.g., things like the structured detectors Lamar did for LOKI?

Reply:
I agree it would be good if we we could generalise the concept of a structured assembly. In many ways our standard PSD tubes could fall into this type of category as well. We already have a special flag so that the instrument view can treat them differently and draw them as one. Perhaps this could feed into that?

Reply:
I think it would be worth taking a look, and I would be happy to have more general concepts. What I am bearing in mind is that there are areas of the existing codebase that explicitly need to identify and work with `RectangularDetector`. The current codebase is peppered with methods like this [example](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument.cpp#L1298)

Reply:
Maybe one option is to store the `RectangularBank` flag to support the case I have linked to above, but also to have a flag that can be used to recognise "classes" of assemblies (`StructuredDetector` etc), for which the processing I have put into `ComponentInfo` would be applicable.